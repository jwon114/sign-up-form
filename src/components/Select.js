import React from 'react';

const Select = ({ register, options, name, noValue, ...rest }) => (
  <select name={name} ref={register} {...rest}>
    <option value="">{noValue}</option>
    {options}
  </select>
);

export default Select;
