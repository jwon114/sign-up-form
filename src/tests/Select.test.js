import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../components/Select';

it('renders correctly', () => {
  const tree = renderer
    .create(<Select 
      register={jest.fn()}
      options={<option key={0} value={1}>test</option>}
      name={'test'}
      noValue={'no value'} />)
    .toJSON()
  expect(tree).toMatchSnapshot();
});
