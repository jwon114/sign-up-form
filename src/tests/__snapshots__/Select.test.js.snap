// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<select
  name="test"
>
  <option
    value=""
  >
    no value
  </option>
  <option
    value={1}
  >
    test
  </option>
</select>
`;
